File: app/src/main/java/com/zawmoehtike/kotlin_workshop/day9/DateExmaple1.java

package com.zawmoehtike.kotlin_workshop.day9;

import java.time.LocalDate;

class DateTimeExample1 {

    public static void main(String[] args) {
        String studentName = "Biber";

        LocalDate today = LocalDate.now(); // today date or current date

        LocalDate yesterday = today.minusDays(1); // yesterday date
        LocalDate tomorrow = yesterday.plusDays(2); // tomorrow date
        LocalDate tomorrow2 = today.plusDays(1); // tomorrow date

        System.out.println("Today date: " + today);
        System.out.println("Yesterday date: " + yesterday);
        System.out.println("Tomorrow date: " + tomorrow);
        System.out.println("Tomorrow2 date: " + tomorrow2);

        // ------------------------------------------------------------------------------------------------------

        LocalDate myBirthday = LocalDate.of(2001, 01, 01); // birthday date

        System.out.println("My birthday is: " + myBirthday);

        // ------------------------------------------------------------------------------------------------------

        String stringDate = "2011-09-01";
        LocalDate stringDateValue = LocalDate.parse(stringDate); // stringDate is converted to stringDateValue

        System.out.println("String Date is: " + stringDate);
        System.out.println("String Date Value is: " + stringDateValue);

        // ------------------------------------------------------------------------------------------------------

        LocalDate kyalKyalBirthday = LocalDate.of(2001, 01, 01);
        LocalDate zawBirthday = LocalDate.of(1991, 01, 01);

        if(kyalKyalBirthday.isAfter(zawBirthday)) { // 2001-01-01 > 1991-01-01
            System.out.println("Kyal Kyal is younger");
        } else {
            System.out.println("Kyal Kyal is older");
        }
    }
}

File: app/src/main/java/com/zawmoehtike/kotlin_workshop/day9/DateExmaple1.kt
package com.zawmoehtike.kotlin_workshop.day9

import android.os.Build
import androidx.annotation.RequiresApi
import java.time.LocalDate
import kotlin.jvm.JvmStatic

internal object DateTimeExample1 {

    @RequiresApi(Build.VERSION_CODES.O)
    @JvmStatic
    fun main(args: Array<String>) {
        val studentName = "Biber"
        val today = LocalDate.now() // today date or current date
        val yesterday = today.minusDays(1) // yesterday date
        val tomorrow = yesterday.plusDays(2) // tomorrow date
        val tomorrow2 = today.plusDays(1) // tomorrow date
        println("Today date: $today")
        println("Yesterday date: $yesterday")
        println("Tomorrow date: $tomorrow")
        println("Tomorrow2 date: $tomorrow2")

        // ------------------------------------------------------------------------------------------------------
        val myBirthday = LocalDate.of(2001, 1, 1) // birthday date
        println("My birthday is: $myBirthday")

        // ------------------------------------------------------------------------------------------------------
        val stringDate = "2011-09-01"
        val stringDateValue =
            LocalDate.parse(stringDate) // stringDate is converted to stringDateValue
        println("String Date is: $stringDate")
        println("String Date Value is: $stringDateValue")

        // ------------------------------------------------------------------------------------------------------
        val kyalKyalBirthday = LocalDate.of(2001, 1, 1)
        val zawBirthday = LocalDate.of(1991, 1, 1)
        if (kyalKyalBirthday.isAfter(zawBirthday)) { // 2001-01-01 > 1991-01-01
            println("Kyal Kyal is younger")
        } else {
            println("Kyal Kyal is older")
        }
    }
}